FindOpenStudioSDK.cmake

```diff
@@ -82,5 +152,53 @@ else()
 
 endif()
 
-get_filename_component(OPENSTUDIO_SDK_PATH "${openstudio_DIR}/../../../" ABSOLUTE)
-set(OPENSTUDIO_SDK_PATH "${OPENSTUDIO_SDK_PATH}" CACHE STRING "This is the path to the root of SDK, under which you'll find bin/, lib/, etc" FORCE)
+get_filename_component(openstudio_ROOT_DIR "${openstudio_DIR}/../../../" ABSOLUTE)
+set(openstudio_ROOT_DIR "${openstudio_ROOT_DIR}" CACHE STRING "This is the path to the root of SDK, under which you'll find bin/, lib/, etc" FORCE)
+
+# Do an extra check?
+find_program(openstudio_EXECUTABLE
+  NAMES openstudio
+  PATHS "${openstudio_ROOT_DIR}/bin/"
+  NO_CMAKE_PATH
+  NO_CMAKE_ENVIRONMENT_PATH
+  NO_SYSTEM_ENVIRONMENT_PATH
+  NO_CMAKE_SYSTEM_PATH
+)
+
+if (NOT EXISTS "${openstudio_EXECUTABLE}")
+  message(AUTHOR_WARNING "Could not find the CLI at ${openstudio_EXECUTABLE}")
+  set_property (CACHE openstudio_EXECUTABLE PROPERTY VALUE "openstudio_EXECUTABLE-NOTFOUND")
+else()
+  # Check the version it returns
+  # executable found must have a specific version
+  execute_process (COMMAND "${openstudio_EXECUTABLE}" "openstudio_version"
+    RESULT_VARIABLE CLI_RESULT
+    OUTPUT_VARIABLE CLI_VERSION
+    ERROR_QUIET
+    OUTPUT_STRIP_TRAILING_WHITESPACE)
+  if (CLI_RESULT)
+    message(AUTHOR_WARNING "Cannot use the openstudio CLI at \"${openstudio_EXECUTABLE}\"")
+  elseif(NOT CLI_VERSION STREQUAL "${OPENSTUDIO_VERSION}${OPENSTUDIO_VERSION_SHA}")
+    execute_process (COMMAND "${openstudio_EXECUTABLE}" -e "puts OpenStudio::openStudioVersion"
+      RESULT_VARIABLE CLI_RESULT
+      OUTPUT_VARIABLE CLI_SHORT_VERSION
+      ERROR_QUIET
+      OUTPUT_STRIP_TRAILING_WHITESPACE)
+    if (CLI_SHORT_VERSION VERSION_EQUAL OPENSTUDIO_VERSION)
+      message(AUTHOR_WARNING
+        "CLI has the same short version (${OPENSTUDIO_VERSION}) but not the same long version."
+        "It has \"${CLI_VERSION}\" versus expected \"${OPENSTUDIO_VERSION}${OPENSTUDIO_VERSION_SHA}\""
+      )
+    else()
+      message(WARNING
+        "CLI has a wrong version: \"${CLI_VERSION}\" versus expected \"${OPENSTUDIO_VERSION}${OPENSTUDIO_VERSION_SHA}\""
+      )
+    endif()
+  else()
+    message(STATUS
+      "Found openstudio CLI with the exact expected long version:\n"
+      "  version: ${CLI_VERSION}\n"
+      "  openstudio_EXECUTABLE: \"${openstudio_EXECUTABLE} \"\n"
+      )
+  endif()
+endif()
```

Comment:
Add some extra checks:
* Check that the CLI exists, and works
* Check that the CLI has the right version
    * If it has the right major.minor.patch but not the right SHA/RC: AUTHOR_WARNING (developpers only)
    * If it has the wrong major.minor.patch: WARNING
 
In all cases, let it fly for now. We could raise to an error in some cases. @macumber